examples/digits.jl

```diff
@@ -56,19 +59,34 @@ end
 
 # Here, we define a schema extension of the `legolas-flux.model` schema.
 # We add our `DigitsConfig` object, as well as the epoch and accuracy.
-const DigitsRow = Legolas.@row("digits.model@1" > "legolas-flux.model@1",
-                               config::DigitsConfig = DigitsConfig(config),
-                               epoch::Union{Missing,Int},
-                               accuracy::Union{Missing,Float32})
+@schema "digits.model" DigitsRow
+
+compat_config(config::DigitsConfigV1) = config
+function compat_config(config::NamedTuple)
+    # This is how these deserialize from Legolas v0.4
+    if haskey(config, 1) && config[1] == "digits-config" && haskey(config, 2) &&
+       config[2] == 1
+        return DigitsConfigV1(config[3])
+    else
+        # We have some other NamedTuple, possibly from an earlier Legolas version. Trying constructing ` DigitsConfigV1`.
+        return DigitsConfigV1(config)
+    end
+end
+
+@version DigitsRowV1 > LegolasFlux.ModelV1 begin
+    config::Union{<:NamedTuple,DigitsConfigV1} = compat_config(config)
+    epoch::Union{Missing,Int}
+    accuracy::Union{Missing,Float32}
+end
```

Comment:
should we also parametrize the row type on the weights?  IIRC that requires re-declaring the field here

```suggestion
@version DigitsRowV1 > LegolasFlux.ModelV1 begin
    # parametrize the row type on the weights type (copied from ModelV1)
    weights::(<:Union{Missing,Weights}) = ismissing(weights) ? missing : Weights(weights)
    config::Union{<:NamedTuple,DigitsConfigV1} = compat_config(config)
    epoch::Union{Missing,Int}
    accuracy::Union{Missing,Float32}
end
```